jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -144,46 +142,62 @@ def _get_marks_for_layers(self, layers):
 
     def _on_subset_delete(self, msg):
         # delete any ShadowSpatialSpectral mark for which either of the spectral or spatial marks
-        # no longer exists
-        spectral_marks = self._get_marks_for_layers(self._get_spectral_subset_layers())
-        spatial_marks = self._get_marks_for_layers(self._get_spatial_subset_layers())
+        # no longer exists by matching the uuid of the msg subset to the uuid of the subsets
+        # in ShadowSpatialSpectral
+        spectral_marks = self._get_marks_for_layers([layer for layer in self._get_spectral_subset_layers() # noqa
+                                                     if layer.layer.uuid == msg.subset.uuid])
+        spatial_marks = self._get_marks_for_layers([layer for layer in self._get_spatial_subset_layers() # noqa
+                                                    if layer.layer.uuid == msg.subset.uuid])
         self.figure.marks = [m for m in self.figure.marks
-                             if not (isinstance(m, ShadowSpatialSpectral) and
-                                     (m.spatial_spectrum_mark in spatial_marks or
-                                      m.spectral_subset_mark in spectral_marks))]
+                             if not (isinstance(m, ShadowSpatialSpectral)
+                                     and msg.subset.uuid in [m.spatial_uuid, m.spectral_uuid])]
```

Comment:
again, this seems to me like it could cause problems when there are multiple highlighting marks and then a subset is deleted.